test/gateway/eip20_cogateway/progress_redeem.js

```diff
@@ -0,0 +1,297 @@
+const Utils = require("../../test_lib/utils.js");
+const web3 = require("../../test_lib/web3.js");
+const EventDecoder = require("../../test_lib/event_decoder");
+
+let MessageStatusEnum = messageBus.MessageStatusEnum;
+contract('EIP20CoGateway.progressRedeem() ', function (accounts) {
+
+  let unlockSecret,
+    amount = new BN(100),
+    beneficiary = accounts[4],
+    eip20CoGateway,
+    messageHash,
+    bountyAmount = new BN(100),
+    utilityToken,
+    facilitator = accounts[1],
+    sender, nonce;
+
+  beforeEach(async function () {
+
+    let gasPrice = new BN(10),
+      gasLimit = new BN(10),
+      hashLockObj = Utils.generateHashLock(),
+      hashLock = hashLockObj.l,
+      owner = accounts[8],
+      intentHash = web3.utils.sha3("dummy");
+
+    sender = owner;
+    unlockSecret = hashLockObj.s;
+    nonce = new BN(1);
+    utilityToken = await MockUtilityToken.new(
+      accounts[9],
+      "",
+      "",
+      18,
+      accounts[2], //organization,
+      {from: owner},
+    );
+
+    eip20CoGateway = await EIP20CoGateway.new(
+      accounts[0],  //value token
+      utilityToken.address,
+      accounts[1],  //state root provider
+      bountyAmount,
+      accounts[2],  //organization
+      accounts[3],  //gateway
+      accounts[10], //burner
+    );
+
+    messageHash = await eip20CoGateway.setMessage.call(
+      intentHash,
+      nonce,
+      gasPrice,
+      gasLimit,
+      sender,
+      hashLock,
+    );
+    await eip20CoGateway.setMessage(
+      intentHash,
+      nonce,
+      gasPrice,
+      gasLimit,
+      sender,
+      hashLock,
+    );
+    await eip20CoGateway.setRedeem(
+      messageHash,
+      beneficiary,
+      amount,
+    );
+    // Set co-gateway to owner so that increase supply can be called.
+    await utilityToken.setCoGatewayAddress(owner);
+    // Send redeem amount to co-gateway.
+    await  utilityToken.increaseSupply(eip20CoGateway.address, amount, {from: owner});
+    // Send bounty to co-gateway.
+    await web3.eth.sendTransaction(
+      {to: eip20CoGateway.address, from: facilitator, value: bountyAmount}
+    );
+    await utilityToken.setCoGatewayAddress(eip20CoGateway.address);
+  });
+
+  it('should emit event on progress redeem', async function () {
+
+    await eip20CoGateway.setOutboxStatus(messageHash, MessageStatusEnum.Declared);
+    let tx = await eip20CoGateway.progressRedeem(messageHash, unlockSecret);
+
+    let event = EventDecoder.getEvents(tx, eip20CoGateway);
+    let eventData = event.RedeemProgressed;
+
+    assert.equal(
+      tx.receipt.status,
+      1,
+      "Receipt status is unsuccessful",
+    );
+    assert.isDefined(
+      event.RedeemProgressed,
+      'Event `RedeemProgressed` must be emitted.',
+    );
+    assert.strictEqual(
+      eventData._messageHash,
+      messageHash,
+      `Actual message hash ${eventData._messageHash} is not equal to expected message hash ${messageHash}`,
```

Comment:
Should we use message like "`Message hash from event.... `" to be more precise? We are using this in other places.